libp2p/connectionMonitor/libp2pConnectionMonitorSimple.go

```diff
@@ -113,21 +156,31 @@ func (lcms *libp2pConnectionMonitorSimple) doReconnectionIfNeeded(netw network.N
 	}
 }
 
-func (lcms *libp2pConnectionMonitorSimple) doReconnection(ctx context.Context) {
+func (lcms *libp2pConnectionMonitorSimple) processLoop(ctx context.Context) {
+	timerCheckConnections := time.NewTimer(durationCheckConnections)
+	timerBetweenReconnectAttempts := time.NewTimer(durationBetweenReconnectAttempts)
 	defer func() {
 		log.Debug("closing the connection monitor main loop")
+		timerCheckConnections.Stop()
+		timerBetweenReconnectAttempts.Stop()
 	}()
 
+	canReconnect := atomic.Flag{}
+	canReconnect.SetValue(true)
 	for {
 		select {
+		case <-timerCheckConnections.C:
+			lcms.checkConnectionsBlocking()
+			timerCheckConnections.Reset(durationCheckConnections)
 		case <-lcms.chDoReconnect:
-		case <-ctx.Done():
-			return
-		}
-		lcms.reconnecter.ReconnectToNetwork(ctx)
-
-		select {
-		case <-time.After(DurationBetweenReconnectAttempts):
+			if !canReconnect.IsSet() {
+				continue
+			}
+			lcms.reconnecter.ReconnectToNetwork(ctx)
+			timerBetweenReconnectAttempts.Reset(durationBetweenReconnectAttempts)
+			canReconnect.SetValue(false)
+		case <-timerBetweenReconnectAttempts.C:
+			canReconnect.SetValue(true)
```

Comment:
forgotten reset call here? How should this process loop work?

Reply:
kept the reset on purpose on the second branch

I was thinking to the following flow, looking to keep only one go routine:
- `timerCheckConnections` added specific for checking the connections which has the reset on its own branch
- then, `timerBetweenReconnectAttempts` is the one that once it fires reconnection is allowed. But, there was no point in resetting the timer right after it fired and set the flag to true, as it won't make any difference(the flag is already true, waiting for a reconnect to network attempt). When that reconnection is done, this timer + its flag will be reset, in order to block a further reconnection attempt